design-proposals/nic-hotplug.md

```diff
@@ -0,0 +1,134 @@
+# Overview
+This KubeVirt design proposal discusses possible implementations for adding -
+and removing - network interfaces from running Virtual Machines, without
+requiring a restart.
+
+## Motivation
+Hot-plug / hot-unplug (add / remove) nics to running VMs is an industry
+standard available in multiple platforms.
+This would help bridge the gap between KubeVirt's feature set and what the user
+base expects from it.
+
+## Goals
+Adding network interfaces to running VMs.
+Removing networking interfaces from running VMs.
+A VM can have multiple interfaces connected to the same (secondary) network(s).
+
+## Non Goals
+Providing multiple connections to the primary cluster network (managed by
```

Comment:
It also would not cover SR-IOV hotplug, right? Since that depends not only on CNI but also on DP. To allow that, we would need to pick approach similar to storage hotplug.

Reply:
Correct. It would deserve a dedicated design document. 

As long as we ensure this design does not make it impossible in the future for sriov, we are OK.

Reply:
Likewise, macvtap interfaces - which are not usable on Openshift - are not in scope. 

I'm listing those as well, since this is an upstream doc.